src/layers/GeoJson.js

```diff
@@ -0,0 +1,157 @@
+import L from 'leaflet';
+import layerMixin from './layerMixin';
+
+// Base class for most vector layers
+export const GeoJson = L.GeoJSON.extend({
+    ...layerMixin,
+
+    options: {
+        style: {
+            weight: 1
+        },
+        highlightStyle: {
+            weight: 3,
+        },
+        resetStyle: {
+            weight: 1,
+        },
+    },
+
+    initialize(options) {
+        L.GeoJSON.prototype.initialize.call(this, options.data, {
+            pane: options.id,
+            pointToLayer: this.pointToLayer.bind(this),
+            ...options,
+        });
+    },
+
+    addLayer(layer) {
+        const { label, hoverLabel, popup } = this.options;
+        const { properties } = layer.feature;
+
+        if (hoverLabel || label) {
+            const tooltip = L.Util.template(hoverLabel || label, properties);
+            layer.bindTooltip(tooltip, {
+                sticky: true,
+            });
+        }
+
+        if (popup && !(popup instanceof Function)) {
+            layer.bindPopup(L.Util.template(popup, properties));
+        }
+
+        L.GeoJSON.prototype.addLayer.call(this, layer);
+    },
+
+    // Use circle markers for point features
+    pointToLayer(geojson, latlng) {
+        this.options.style.pane = this.options.pane;
+        return new L.CircleMarker(latlng, this.options.style);
+    },
+
+    setOpacity(opacity) {
+        this.setStyle({
+            opacity,
+            fillOpacity: opacity,
+        });
+    },
+
+    findById(id) {
+        for (const i in this._layers) {
+            // eslint-disable-line
+            if (this._layers[i].feature.id === id) {
+              return this._layers[i];
+            }
+        }
+
+        return null;
+    },
+
+    onAdd(map) {
+        const {
+            opacity,
+            isVisible,
+            onClick,
+            onRightClick,
+            highlightStyle,
+        } = this.options;
+
+        L.GeoJSON.prototype.onAdd.call(this, map);
```

Comment:
Why do we need to use `.call` for these methods?

Reply:
See: https://leafletjs.com/examples/extending/extending-1-classes.html#methods-of-the-parent-class